### PR TITLE
switching to persistent journal possible without rsyslog restart, 

### PR DESCRIPTION
**Pull request by jvymazal**
bumped corresponding systemd version dependency

### CODE REVIEW

**Comment by jvymazal** [comment]
@rgerhards  please, would this change be viable?

**Comment by rgerhards** [comment]
I just had a look, but need to have a closer one. The change looks good on first view. However, you should squash the commits to get a clean commit history. I guess it's actually just a single commit from a logical PoV.

**Comment by jvymazal** [comment]
I will squash the commits once the change is approved, thanks for your review.

File: plugins/imjournal/imjournal.c [hunk 1]
```diff
@@ -472,12 +496,28 @@ pollJournal(void)
...
+		char* tmp = cs.stateFile;
+		cs.stateFile = NULL;
+		closeJournal(j);
+		cs.stateFile = tmp;
+
+		iRet = openJournal(j);
+		if (iRet != RS_RET_OK) {
+			char errStr[256];
+			rs_strerror_r(errno, errStr, sizeof(errStr));
+			errmsg.LogError(0, RS_RET_IO_ERROR,
```
Comment (rgerhards) [inline plugins/imjournal/imjournal.c]: just a small way to improve: we made ```LogError``` available without the need to prefix ```errmsg.``` - it's a little bit more efficient. Of course, all is well, but the new method is the preferred one.

File: plugins/imjournal/imjournal.c [hunk 2]
```diff
@@ -472,12 +496,28 @@ pollJournal(void)
...
+				"sd_journal_open() failed: '%s'", errStr);
+			ABORT_FINALIZE(RS_RET_ERR);
+		}
 
+		iRet = loadJournalState();
+		errmsg.LogError(0, RS_RET_OK, "imjournal: journal reloaded...");
```
Comment (rgerhards) [inline plugins/imjournal/imjournal.c]: That's probably better to do as informational: ```LogMsg(0, RS_RET_OK, "imjournal: journal reloaded...");```

**Comment by rgerhards** [comment]
@jvymazal oh, one thing: it would be good if you could amend the commit text with a good description of the actual motivation for the change and what it means to the average user.

**Comment by rgerhards** [comment]
@jvymazal it looks like the journal version now requested is too recent for almost all distros. Can you work on graceful degradation if that version cannot be found?

**Comment by jvymazal** [comment]
Ok, I will try to make it easily conditionable (at least temporarily before 234 systemd gets more widespread) and open a new PR with it.

**Comment by rgerhards** [comment]
That's very good. I'll keep the current PR until the weekend. If the problems become to big, I'll temporarily revert it.

**Pull request merged**